uvicorn/protocols/websockets/wsproto_impl.py

```diff
@@ -59,25 +66,42 @@ def connection_made(self, transport):
         self.scheme = "wss" if is_ssl(transport) else "ws"
 
     def connection_lost(self, exc):
+        if exc is not None:
+            self.queue.put_nowait({"type": "websocket.disconnect"})
         self.connections.remove(self)
 
     def eof_received(self):
         pass
 
     def data_received(self, data):
-        self.conn.receive_bytes(data)
+        # Push data. Can raise RemoteProtocolError (previously resulted in Fail event)
+        try:
+            self.conn.receive_data(data)
+        except RemoteProtocolError as err:
+            if err.event_hint is not None:
+                self.transport.write(self.conn.send(err.event_hint))
+                self.transport.close()
+            else:
+                self.handle_no_connect(events.CloseConnection())
+        else:
+            self.handle_events()
+
+    def handle_events(self):
         for event in self.conn.events():
-            if isinstance(event, wsproto.events.ConnectionRequested):
+            if isinstance(event, events.Request):
                 self.handle_connect(event)
-            elif isinstance(event, wsproto.events.TextReceived):
+            elif isinstance(event, events.TextMessage):
                 self.handle_text(event)
-            elif isinstance(event, wsproto.events.BytesReceived):
+            elif isinstance(event, events.BytesMessage):
                 self.handle_bytes(event)
-            elif isinstance(event, wsproto.events.ConnectionFailed):
+            elif isinstance(event, events.RejectConnection):
+                self.handle_no_connect(event)
+            elif isinstance(event, events.RejectData):
                 self.handle_no_connect(event)
-            elif isinstance(event, wsproto.events.ConnectionClosed):
+            elif isinstance(event, events.CloseConnection):
                 self.handle_close(event)
-            elif isinstance(event, wsproto.events.PingReceived):
+                break  # todo: right?
```

Comment:
@tomchristie what do you think?

Reply:
We probably don't need it there. `wsproto` ought to take take of giving us a consistent stream of messages. I'd drop the `break` assuming the tests still pass.